src/sortedcache.h

```diff
@@ -155,26 +157,29 @@ int git_sortedcache_remove(git_sortedcache *sc, size_t pos);
  */
 
 /* Lock sortedcache for read */
-int git_sortedcache_rlock(git_sortedcache *sc);
+GIT_WARN_UNUSED_RESULT int git_sortedcache_rlock(git_sortedcache *sc);
 
 /* Unlock sorted cache when done with read */
 void git_sortedcache_runlock(git_sortedcache *sc);
 
 /* Lookup item by key - returns NULL if not found */
-void *git_sortedcache_lookup(const git_sortedcache *sc, const char *key);
+GIT_WARN_UNUSED_RESULT void *git_sortedcache_lookup(
```

Comment:
I like `GIT_WARN_UNUSED_RESULT` for things that return errors that really must be paid attention to, for example things that allocate.  I'm less bullish on things like this.  One on hand, I think that I understand the motivation: _why are you calling this function_ if you don't care about the result?  It's side effect free.  So its _only purpose in life_ is to compute a result for you.  Calling it and ignoring the result is just a waste of cycles.

🤔

I feel like we should draw up some simple guidelines about where and when we use `GIT_WARN_UNUSED_RESULT`...  It feels like you're proposing a place where we land on it _almost always_.  Is that roughly accurate?

Reply:
i could also be okay with restricting its use to only places that allocate or grab a mutex (that case is also pretty scary).

if/when another clang-tidy solution exists where everything is checked except for a few allow-listed functions, things'll be less verbose.